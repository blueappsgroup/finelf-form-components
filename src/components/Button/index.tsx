import React from 'react'
import styled from 'styled-components'

import { ButtonProps } from '../../types'

type StyledProps = {
  theme: {
    submitButtonBgColor: string
    submitButtonTextColor: string
  }
}

const StyledButton = styled.button`
  background: ${(props: StyledProps): string =>
    props.theme.submitButtonBgColor};
  border-radius: 4px;
  border: none;
  color: ${(props: StyledProps): string => props.theme.submitButtonTextColor};
  outline: 0;
  letter-spacing: 0.5px;
  font-style: normal;
  font-weight: 800;
  font-size: 20px;
  line-height: 16px;
  padding: 28px;
  margin-top: 20px;
`

const Button: React.FC<ButtonProps> = ({ text }) => (
  <StyledButton type="submit">{text}</StyledButton>
)

export default Button
