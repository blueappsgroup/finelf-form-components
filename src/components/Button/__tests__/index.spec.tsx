import * as React from 'react'
import { render } from '@testing-library/react'
import Button from '../'

describe('<Button />', () => {
  it('matches snapshot', () => {
    const documentBody = render(<Button text="Submit" />)
    const { baseElement } = documentBody
    expect(documentBody.getByText('Submit')).toBeTruthy()
    expect(baseElement).toMatchSnapshot()
  })

  it('handles onClick Event', () => {
    const handleClick = jest.fn()
    const documentBody = render(<Button text="Submit" onClick={handleClick} />)
    const buttonBody = documentBody.getByText('Submit')
    buttonBody.click()
    expect(handleClick).toBeCalledTimes(1)
  })
})
