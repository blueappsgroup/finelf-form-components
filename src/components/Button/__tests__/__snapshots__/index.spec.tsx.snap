// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> matches snapshot 1`] = `
<body>
  <div>
    <button
      class="sc-bdnylx bMLZUh"
      type="button"
    >
      Submit
    </button>
  </div>
</body>
`;
